architecture/core_concepts/routes.adoc

```diff
@@ -12,41 +12,87 @@ toc::[]
+need to modify its DNS records independently to resolve to the node that
+is running the router.
+
+Each route consists of a route name (limited to 63 characters), service selector,   
+and (optionally) security configuration.
 
 == Routers
 
-An OpenShift administrator can deploy _routers_ in an OpenShift cluster, which
-enable routes link:../../dev_guide/routes.html#creating-routes[created by
-developers] to be used by external clients. The routing layer in OpenShift is
-pluggable, and two available link:#available-router-plug-ins[router plug-ins]
-are provided and supported by default.
+An {product-title} administrator can deploy _routers_ to nodes in an
+{product-title} cluster, which enable routes
+xref:../../dev_guide/routes.adoc#creating-routes[created by developers]
+to be used by external clients.
+
+
+The routing layer in {product-title} is
+pluggable, and two available xref:#available-router-plug-ins[router plug-ins]
+are provided and supported.
 
 ifdef::openshift-enterprise,openshift-origin[]
 [NOTE]
 ====
-See the link:../../install_config/install/deploy_router.html[Installation and
+See the xref:../../install_config/install/deploy_router.adoc#[Installation and
 Configuration] guide for information on deploying a router.
 ====
 endif::[]
 
-OpenShift routers provide external host name mapping and load balancing
-to link:pods_and_services.html#services[services] over protocols that
+A router uses the service selector to find the service and the endpoints backing
+the service. Service-provided load balancing is bypassed and replaced with the
+router's own load balancing. Routers watch the cluster API and automatically
+update their own configuration according to any relevant changes to the IP
+addresses of pods that provide the service. Custom routers can be deployed to
+communicate modifications of API objects to an external routing solution.
+
+In order to reach a router in the first place, requests for host names must
+resolve via DNS to a router or set of routers. The suggested method is to define
+a cloud domain with a wildcard DNS entry pointing to a virtual IP (VIP) backed by
+multiple router instances on designated nodes.
+
+Routes can be
+xref:../../architecture/core_concepts/routes.adoc#router-sharding[sharded]
+among the set of routers. Administrators can set up sharding on a cluster
+wide basis and users can set up sharding for the namespace in their project.
+Sharding allows the operator to define multiple router groups. Each router
+in the group serves only a subset of traffic.
+
+Router VIP configuration is described in
+ifdef::openshift-enterprise,openshift-origin[]
+xref:../../admin_guide/high_availability.adoc#configuring-a-highly-available-routing-service[High Availability].
+endif::[]
+ifdef::openshift-dedicated[]
+the https://docs.openshift.com/enterprise/3.1/admin_guide/high_availability.html#configuring-a-highly-available-routing-service[{product-title} Enterprise Cluster Administration documentation].
```

Comment:
This `{product-title}` should actually just be "OpenShift" since it's referring to a specific distro name outside of its own. The way it is currently, it would get transformed to "OpenShift Dedicated Enterprise Cluster Administration".